gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-client/src/test/java/gobblin/service/FlowStatusTest.java

```diff
@@ -0,0 +1,232 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.service;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.inject.Binder;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.name.Names;
+import com.linkedin.restli.server.resources.BaseResource;
+
+import gobblin.config.ConfigBuilder;
+import gobblin.metrics.event.TimingEvent;
+import gobblin.restli.EmbeddedRestliServer;
+import gobblin.service.monitoring.FlowStatusGenerator;
+import gobblin.service.monitoring.JobStatusRetriever;
+
+@Test(groups = { "gobblin.service" }, singleThreaded=true)
+public class FlowStatusTest {
+  private FlowStatusClient _client;
+  private EmbeddedRestliServer _server;
+  private List<gobblin.service.monitoring.JobStatus> _jobStatusList;
+  private Joiner messageJoiner;
+
+  class TestJobStatusRetriever extends JobStatusRetriever {
+    @Override
+    public Iterator<gobblin.service.monitoring.JobStatus> getJobStatusesForFlowExecution(String flowName,
+        String flowGroup, long flowExecutionId) {
+      return _jobStatusList.iterator();
+    }
+
+    @Override
+    public long getLatestExecutionIdForFlow(String flowName, String flowGroup) {
+      return 0;
+    }
+  }
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    ConfigBuilder configBuilder = ConfigBuilder.create();
+
+    JobStatusRetriever jobStatusRetriever = new TestJobStatusRetriever();
+    final FlowStatusGenerator flowStatusGenerator =
+        FlowStatusGenerator.builder().jobStatusRetriever(jobStatusRetriever).build();
+
+    Injector injector = Guice.createInjector(new Module() {
+       @Override
+       public void configure(Binder binder) {
+         binder.bind(FlowStatusGenerator.class).annotatedWith(Names.named(FlowStatusResource.FLOW_STATUS_GENERATOR_INJECT_NAME))
+             .toInstance(flowStatusGenerator);
+       }
+    });
+
+    _server = EmbeddedRestliServer.builder().resources(
```

Comment:
The embedded server will find an open port. FS working path for each test should be set up using a generated temp path or path based on path name to avoid collision.